internal/datanode/l0_compactor.go

```diff
@@ -162,8 +162,8 @@ func (t *levelZeroCompactionTask) compact() (*datapb.CompactionPlanResult, error
 	}
 
 	var resultSegments []*datapb.CompactionSegment
-	// TODO
-	if hardware.GetFreeMemoryCount() < uint64(totalSize)*20 {
+
+	if float64(hardware.GetFreeMemoryCount())*paramtable.Get().DataNodeCfg.L0BatchMemoryRatio.GetAsFloat() < float64(totalSize) {
```

Comment:
make sure that, the old condition means that if totalSize reach 5% of free memory, it will turn to linear process. and the new condition change to 20%?

Reply:
changed back to original value now